File: ruby/easy/valid_parantheses.rb
# @param {String} s
# @return {Boolean}
def is_valid(s)
  map = { '(': ')', '{': '}', '[':']' };
  stack = []
  s.each_char do |c|
    if(map[c.to_sym])
      stack.push(c)
    else
      return false if(stack.empty? || map[stack.pop().to_sym] != c)
    end
  end
  return stack.empty?
end

File: javascript/easy/validParantheses.js
/**
 * @param {string} s
 * @return {boolean}
 */
var isValid = function(s) {
    let stack = [];
    let map = { '(': ')', '[': ']', '{': '}' };
    for (c of s) {
        if(map[c] !== undefined) {
            stack.push(c);
        } else {
            if(stack.length == 0 || (map[stack.pop()] !== c)) {
                return false;
            }
        }
    }
    return stack.length == 0;
};